src/bp/migrations/v12_18_1-1614716851-fixed_agent_role_rules.ts

```diff
@@ -0,0 +1,46 @@
+import * as sdk from 'botpress/sdk'
+import { MigrationOpts } from 'core/services/migration'
+import { WorkspaceService } from 'core/services/workspace-service'
+import { TYPES } from 'core/types'
+import _ from 'lodash'
+
+const AGENT_ROLE_ID = 'agent'
+const RULES_TO_ADD = [
+  { res: 'admin.collaborators.*', op: '-r' },
+  { res: 'admin.logs.*', op: '-r' },
+  { res: 'admin.roles.*', op: '-r' }
+]
+
+const migration: sdk.ModuleMigration = {
+  info: {
+    description: 'Alter agent role rules in order to hide sections in admin ui',
+    target: 'core',
+    type: 'config'
+  },
+  up: async ({ inversify }: MigrationOpts): Promise<sdk.MigrationResult> => {
+    const service = inversify.get<WorkspaceService>(TYPES.WorkspaceService)
+    const workspaces = await service.getWorkspaces()
+
+    const alteredWorkspaces = workspaces.map(ws => {
+      const agentRole = ws.roles.find(r => r.id === AGENT_ROLE_ID)
+      if (!agentRole || _.intersectionBy(agentRole.rules, RULES_TO_ADD, 'res').length === 3) {
+        return ws
+      }
```

Comment:
Is it intended behavior that a role that has already defined the same permissions but with different `op` stayed the same?

Reply:
I'm sorry I don't understand your question. What I do in the right part of the ||  is a simple logic to test if that migration has already been run.

Reply:
Ah ok it's all good then!